templates/Dockerfile-deb.tmpl

```diff
@@ -115,6 +116,17 @@ RUN wget https://github.com/Kitware/CMake/releases/download/v3.22.2/cmake-3.22.2
     rm -f cmake-3.22.2.tar.gz && \
     rm -rf cmake-3.22.2
 
+# install pyenv and python 3.8 to be able to execute tools scripts
+RUN git clone https://github.com/pyenv/pyenv.git ~/.pyenv && \
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bashrc && \
+    echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bashrc && \
+    echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n eval "$(pyenv init -)"\nfi' >> ~/.bashrc
+
+# install python 3.8 and set it as default. Reloading the bashrc did not worked for ubuntu \
+# therefore I need to export the path explicitly here.
+RUN source ~/.bashrc && export PYENV_ROOT="$HOME/.pyenv" && export PATH="$PYENV_ROOT/bin:$PATH" && \
+    pyenv install 3.8.16 && pyenv global 3.8.16
+
```

Comment:
```suggestion
# Install pyenv
ARG PYTHON_VERSION=3.8.16
RUN set -ex \
    && curl https://pyenv.run | bash \
    && pyenv update \
    && pyenv install $PYTHON_VERSION \
    && pyenv global $PYTHON_VERSION \
    && pyenv rehash
```

This online shell script (https://pyenv.run) actually sets the necessary variables, downloads the git repo**s** when run.

You can apply a similar patch for rpm as well. It will make it unnecessary to use bash.

Reply:
You can also move the `ARG` line to earlier in the file.

Reply:
@hanefi I think it is not good to rely on some domain name we don't know when we look at in security perspective. 
When I make a whois search, I see that this domain name owner is RadiaSoft LLC which is an unknown company.
However, parameterization of Python version is good idea. I need to handle that

Reply:
How about the following?

`curl -s -S -L https://raw.githubusercontent.com/pyenv/pyenv-installer/master/bin/pyenv-installer | bash`

That link that is hosted by RadiaSoft is actually a one-liner that runs the most up-to-date pyenv-installer script. Do you think that it is safe to install a file from a well known Github repo and execute it?

Reply:
See the following docs from pyenv-installer: https://github.com/pyenv/pyenv-installer#using-pyenv-installer-on-travis-ci

Reply:
Changed as you suggested